src/test/java/it/pagopa/ecommerce/payment/requests/utils/confidential/ConfidentialTest.java

```diff
@@ -0,0 +1,86 @@
+package it.pagopa.ecommerce.payment.requests.utils.confidential;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import it.pagopa.ecommerce.payment.requests.utils.confidential.domain.Confidential;
+import it.pagopa.ecommerce.payment.requests.utils.confidential.domain.Email;
+import it.pagopa.generated.pdv.v1.api.TokenApi;
+import it.pagopa.generated.pdv.v1.dto.PiiResourceDto;
+import it.pagopa.generated.pdv.v1.dto.TokenResourceDto;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import reactor.core.publisher.Mono;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+
+@ExtendWith(MockitoExtension.class)
+class ConfidentialTest {
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final TokenApi personalDataVaultClient = Mockito.mock(TokenApi.class);
+
+    private final ConfidentialDataManager confidentialDataManager = new ConfidentialDataManager(
+            personalDataVaultClient
+    );;
+
+    @Test
+    void confidentialJsonRepresentationIsOK() throws JsonProcessingException {
+        Email email = new Email("foo@example.com");
+
+        /* preconditions */
+        Mockito.when(personalDataVaultClient.saveUsingPUT(any()))
+                .thenReturn(Mono.just(new TokenResourceDto().token(UUID.randomUUID())));
+
+        /* test */
+        Confidential<Email> confidentialEmail = this.confidentialDataManager.encrypt(email)
```

Comment:
You can use StepVerifier and not perform a blocking operation
```suggestion
        StepVerifier.create(confidentialDataManager.decrypt)
          .consumeNextWith(next -> {"assert to execute"})
          .verifyComplete();`
```